### PR TITLE
[BPK-2423] Implemented Floating Action Button

### PR DESCRIPTION
**Pull request by bvitaliyg**
### **DONT MERGE YET, REVIEW ONLY**

**Api/Implementation** review only, the tests and documentation will be added when it's approved.

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/master/CONTRIBUTING.md)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

**Comment by georgegillams** [comment]
Thank you for your contribution.
You can use the following dependency to test your application while we review your pull request:

`com.github.Skyscanner:backpack-android:37459b9a2bbcfbc98cddffb716dc2ca556b3b76d`

File: Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.util.BpkTheme
+import net.skyscanner.backpack.util.createContextThemeWrapper
+import net.skyscanner.backpack.util.darken
+
+internal open class BpkFab @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : AppCompatButton(createContextThemeWrapper(context, attrs, R.attr.bpkFabStyle), attrs, defStyleAttr) {
```
Comment (amolgupta) [inline Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt]: Can we not extend from `FloatingActionButton`. Would save a lot of effort and give all the other functionalities of the component.

Comment (bvitaliyg) [reply Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt]: The general idea here is to migrate all (BpkButton, BpkFab) to the same base class and have the same functionality and api -- what's why it extends AppCompatButton.

File: app/src/androidTest/java/net/skyscanner/backpack/fab/BpkFabTest.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  @Test
+  fun screenshotTestFab_withTheme() {
+    val fab = BpkFab(createThemedContext(testContext))
+    fab.icon = ContextCompat.getDrawable(testContext, R.drawable.bpk_search)
+    snap(fab)
+  }
+
+  @Test
+  fun screenshotTestFab_rtl() {
+    fab.layoutDirection = View.LAYOUT_DIRECTION_RTL
```
Comment (amolgupta) [inline app/src/androidTest/java/net/skyscanner/backpack/fab/BpkFabTest.kt]: The RTL does not seem to be doing anything. You could add a `drawable-rtl` for the icon to test.

Comment (bvitaliyg) [reply app/src/androidTest/java/net/skyscanner/backpack/fab/BpkFabTest.kt]: The point here is just to make sure that nothing is broken is RTL. It does not test the icon's mirroring.

**Comment by georgegillams** [comment]
Thank you for your contribution.
You can use the following dependency to test your application while we review your pull request:

`com.github.Skyscanner:backpack-android:00e2b149bd84a2bc72e1e5570c0223196148eae7`

**Pull request merged**